k8s/k8s-workload-registrar/README.md

```diff
@@ -0,0 +1,716 @@
+
+Something similar happens with the pod entries, but this time the pod UID where the workload is running is stored in the selectors, instead of the node UID.
+
+If we now focus our attention on the SPIFFE IDs assigned to the workloads, we see that it takes the form of *spiffe://\<TRUSTDOMAIN\>/\<ANNOTATIONVALUE\>*. By using Annotation Based workload registration, it is possible to freely set the SPIFFE ID path. In this case, for the workload, we set the annotation value to *example-workload*.
+
+Obtain the registrar logs by issuing: 
+
+```console
+kubectl logs statefulset/spire-server -n spire -c k8s-workload-registrar | grep "Created entry"
+```
+
+This will show that the registrar created the 3 entries into the SPIRE Server.
+
+In addition to the SPIRE entries, the registrar in this mode is configure to create the corresponding custom resources. Let's check for this using a Kubernetes native command such as:
+
+```console
+$ kubectl get spiffeids -n spire
+```
+
+This command will show the custom resources for each one of the pods:
+
+```console
+NAME                               AGE
+example-cluster-control-plane      24m
+example-workload-5bffcd75d-stl5w   24m
+spire-agent-r86rz                  24m
+```
+
+## Pod deletion
+
+As in the previous modes, if we delete the workload deployment, we will see that its corresponding registration entry will be deleted too. Let's check it by running the command to delete the workload pod: 
+
+```console
+$ kubectl delete deployment/example-workload -n spire
+```
+
+And now, check the registration entries in the SPIRE Server by executing:
+
+```console
+$ kubectl exec -it statefulset/spire-server -n spire -c spire-server -- bin/spire-server entry show -registrationUDSPath /tmp/spire-server/private/api.sock
+```
+
+The output should look like:
+
+```console
+Found 2 entries
+Entry ID         : ...
+SPIFFE ID        : spiffe://example.org/k8s-workload-registrar/example-cluster/node/example-cluster-control-plane
+Parent ID        : spiffe://example.org/spire/server
+Revision         : 1
+TTL              : default
+Selector         : k8s_psat:agent_node_uid:08990bfd-3551-4761-8a1b-2e652984ffdd
+Selector         : k8s_psat:cluster:example-cluster
+
+Entry ID         : ...
+SPIFFE ID        : spiffe://example.org/testing/agent
+Parent ID        : spiffe://example.org/k8s-workload-registrar/example-cluster/node/example-cluster-control-plane
+Revision         : 1
+TTL              : default
+Selector         : k8s:node-name:example-cluster-control-plane
+Selector         : k8s:ns:spire
+Selector         : k8s:pod-uid:538886bb-48e1-4795-b386-10e97f50e34f
+DNS name         : spire-agent-jzc8w
+```
+
+The only entries that should exist now are the ones that match the node and the SPIRE agent, because the workload one was deleted by the registrar, something that we can check if we examine the registrar logs, but this time looking for the keyword "Deleted entry".
```

Comment:
```suggestion
The only entries that should exist now are the ones that match the node and the SPIRE Agent, because the workload one was deleted by the registrar, something that we can check if we examine the registrar logs, but this time looking for the keyword "Deleted entry":
```